Sources/SwiftCliCore/InputValidator.swift

```diff
@@ -15,8 +17,8 @@ public class InputValidator {
             }
         }
 
-        var patternNestedArray =  pattern.map { NestedArrayBuilder().arrayMaker(row: $0) }
-        if (knitFlat == true) {
+        var patternNestedArray =  lowercaseNormalizedPattern.map { NestedArrayBuilder().arrayMaker(row: $0) }
```

Comment:
Similar refactor to inject `NestedArrayBuilder` into `InputValidator` rather than initialize it directly inside this method.

Reply:
Sounds good! One question I do have though : Would I then need to inject `NestedArrayBuilder` into the class that calls `InputValidator`, which is `Chartify`, and the class that calls that, which is `main`? Or can I simply create a new instance and pass it in when I call `InputValidator` in `Chartify`? It seems like the latter would be sufficient for mocking, is that right?

Reply:
If you inject `NestedArrayBuilder` into `InputValidator`, you wouldn't need to inject it into `Chartify`. If for whatever reason `Chartify` needed to access a `NestedArrayBuilder`, you would have a `getNestedArrayBuilder` method in `InputValidator` which would return the `NestedArrayBuilder` object that `InputValidator` was initialized with.

Not sure if that addresses your question so let me know if that misses the mark.

Reply:
Update: I realized I needed to put everything in from the `main` file if I was going to do a dependency injection. At least I'm fairly sure that's the case.

Reply:
Yes, so you would have presumably in your `main`:

```swift
let nestedArrayBuilder = NestedArrayBuilder()
let patternNormalizer = PatternNormalizer()
let inputValidator = InputValidator(nestedArrayBuilder: nestedArrayBuilder, patternNormalizer: patternNormalizer)

let chartConstructor = ChartConstructor()
let fileValidator = FileValidator()
let chartify = Chartify(inputValidator: inputValidator, chartConstructor: chartConstructor, fileValidator: fileValidator)
```

But the dependency injection is only happening for `InputValidator` and `Chartify` in the above code, and you're not injecting `NestedArrayBuilder` into `Chartify`.

Reply:
This also has other advantages, like setting `knitFlat` as a property when you initialize `InputValidator` instead of having to pass it down via `Chartify`.